<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1226</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="saveBtn">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>110</y>
      <width>141</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Save Image</string>
    </property>
   </widget>
   <widget class="QPushButton" name="loadBtn">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>60</y>
      <width>141</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Load Image</string>
    </property>
   </widget>
   <widget class="QPushButton" name="deBlurBtn">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>60</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Deblur Image</string>
    </property>
   </widget>
   <widget class="QPushButton" name="grayScaleBtn">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>110</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Grayscale Image</string>
    </property>
   </widget>
   <widget class="QPushButton" name="flipCwBtn">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>160</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Flip Image (CW)</string>
    </property>
   </widget>
   <widget class="QPushButton" name="flipCcwBtn">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>210</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Flip Image (CCW)</string>
    </property>
   </widget>
   <widget class="QPushButton" name="cropBtn">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>160</y>
      <width>141</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Crop Image</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="colorBalanceValue">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>265</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QProgressBar" name="progressBar">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>320</y>
      <width>1051</width>
      <height>23</height>
     </rect>
    </property>
    <property name="value">
     <number>24</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="saturationValue">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>100</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QSpinBox" name="contrastValue">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>140</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QSpinBox" name="rotateValue">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>180</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QSpinBox" name="blurValue">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>220</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="mirrorBtn">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>60</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Mirror Image</string>
    </property>
   </widget>
   <widget class="QPushButton" name="reverseBtn">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>110</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Reverse Color</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>10</y>
      <width>441</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'.AppleSystemUIFont'; font-size:13pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p align=&quot;justify&quot; style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:12pt; font-weight:600;&quot;&gt;	&lt;/span&gt;&lt;span style=&quot; font-size:14pt; font-weight:600;&quot;&gt;Edit&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="margin">
     <number>0</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>10</y>
      <width>81</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt; font-weight:600;&quot;&gt;File&lt;br/&gt;&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="textFormat">
     <enum>Qt::RichText</enum>
    </property>
    <property name="margin">
     <number>10</number>
    </property>
    <property name="openExternalLinks">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="addNoiseBtn">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>160</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Add Noise</string>
    </property>
   </widget>
   <widget class="QPushButton" name="detectBtn">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>210</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Detect Edges</string>
    </property>
   </widget>
   <widget class="QComboBox" name="rgbValue">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>270</y>
      <width>161</width>
      <height>26</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Red</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Green</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Blue</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="applyBtn">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>990</x>
      <y>120</y>
      <width>113</width>
      <height>32</height>
     </rect>
    </property>
    <property name="palette">
     <palette>
      <active>
       <colorrole role="Button">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>23</red>
          <green>168</green>
          <blue>26</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Midlight">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Base">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>23</red>
          <green>168</green>
          <blue>26</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Window">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>23</red>
          <green>168</green>
          <blue>26</blue>
         </color>
        </brush>
       </colorrole>
      </active>
      <inactive>
       <colorrole role="Button">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>23</red>
          <green>168</green>
          <blue>26</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Midlight">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Base">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>23</red>
          <green>168</green>
          <blue>26</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Window">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>23</red>
          <green>168</green>
          <blue>26</blue>
         </color>
        </brush>
       </colorrole>
      </inactive>
      <disabled>
       <colorrole role="Button">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>23</red>
          <green>168</green>
          <blue>26</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Midlight">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Base">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>23</red>
          <green>168</green>
          <blue>26</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Window">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>23</red>
          <green>168</green>
          <blue>26</blue>
         </color>
        </brush>
       </colorrole>
      </disabled>
     </palette>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(23, 168, 26);
</string>
    </property>
    <property name="text">
     <string>Apply</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
    <property name="autoDefault">
     <bool>false</bool>
    </property>
    <property name="default">
     <bool>false</bool>
    </property>
    <property name="flat">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="resetBtn">
    <property name="geometry">
     <rect>
      <x>990</x>
      <y>170</y>
      <width>113</width>
      <height>32</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(246, 0, 11);</string>
    </property>
    <property name="text">
     <string>Reset</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>60</y>
      <width>181</width>
      <height>241</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QCheckBox" name="brightness">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>141</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Change Brightness </string>
     </property>
    </widget>
    <widget class="QCheckBox" name="saturation">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>141</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Change Saturation</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="contrast">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>141</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Change Contrast</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="rotate">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>130</y>
       <width>141</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Rotate Image</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="blur">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>170</y>
       <width>161</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Blur Image</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="colorBalance">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>210</y>
       <width>161</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Change Color Balance</string>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="inputFrame">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>360</y>
      <width>511</width>
      <height>341</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border-color: rgb(169, 236, 160);</string>
    </property>
    <property name="text">
     <string>                                                       INPUT IMAGE</string>
    </property>
   </widget>
   <widget class="QLabel" name="outputFrame">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>360</y>
      <width>511</width>
      <height>341</height>
     </rect>
    </property>
    <property name="text">
     <string>                                                       OUTPUT IMAGE</string>
    </property>
   </widget>
   <widget class="QLabel" name="messageText">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>295</y>
      <width>651</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Message :</string>
    </property>
   </widget>
   <widget class="QLabel" name="percentageBox">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>320</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>%0</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="brightnessValue">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>60</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1226</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuMain_Window">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad_Image"/>
    <addaction name="actionSave_Image"/>
    <addaction name="actionShow_Image"/>
    <addaction name="actionCrop_Image"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionBlur_Image"/>
    <addaction name="actionDeblur_Image"/>
    <addaction name="actionMirror_Image"/>
    <addaction name="separator"/>
    <addaction name="actionReverse_Color"/>
    <addaction name="actionGrayScale_Image"/>
    <addaction name="actionAdd_Noise"/>
    <addaction name="separator"/>
    <addaction name="actionFlip_Image_CW"/>
    <addaction name="actionFlip_Image_CCW"/>
    <addaction name="actionDetect_Edges"/>
   </widget>
   <addaction name="menuMain_Window"/>
   <addaction name="menuEdit"/>
  </widget>
  <action name="actionBlur_Image">
   <property name="text">
    <string>Blur Image</string>
   </property>
  </action>
  <action name="actionLoad_Image">
   <property name="text">
    <string>Load Image</string>
   </property>
  </action>
  <action name="actionSave_Image">
   <property name="text">
    <string>Save Image</string>
   </property>
  </action>
  <action name="actionShow_Image">
   <property name="text">
    <string>Show Image</string>
   </property>
  </action>
  <action name="actionDeblur_Image">
   <property name="text">
    <string>Deblur Image</string>
   </property>
  </action>
  <action name="actionMirror_Image">
   <property name="text">
    <string>Mirror Image</string>
   </property>
  </action>
  <action name="actionReverse_Color">
   <property name="text">
    <string>Reverse Color</string>
   </property>
  </action>
  <action name="actionGrayScale_Image">
   <property name="text">
    <string>GrayScale Image</string>
   </property>
  </action>
  <action name="actionFlip_Image_CW">
   <property name="text">
    <string>Flip Image (CW)</string>
   </property>
  </action>
  <action name="actionFlip_Image_CCW">
   <property name="text">
    <string>Flip Image (CCW)</string>
   </property>
  </action>
  <action name="actionDetect_Edges">
   <property name="text">
    <string>Detect Edges</string>
   </property>
  </action>
  <action name="actionAdd_Noise">
   <property name="text">
    <string>Add Noise</string>
   </property>
  </action>
  <action name="actionCrop_Image">
   <property name="text">
    <string>Crop Image</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
